bin/reference_tools/update_bed.py

```diff
@@ -0,0 +1,450 @@
+
+    clinvar_in_common = new_clinvar_keys.intersection(old_clinvar_keys)
+    clinvar_new_added = new_clinvar_keys.difference(old_clinvar_keys)
+
+    new_to_add_bed_rows = get_bed_intersect(
+        str(new_clinvar_tmp_bed), str(final_bed_path)
+    )
+    old_to_remove_bed_rows = get_bed_intersect(
+        str(old_clinvar_tmp_bed), str(final_bed_path)
+    )
+    if not keep_tmp:
+        new_clinvar_tmp_bed.unlink()
+        old_clinvar_tmp_bed.unlink()
+
+    print(f"Clinvar in common between versions: {len(clinvar_in_common)}")
+    print(
+        f"Added new (unique targets): {len(clinvar_new_added)} ({len(new_to_add_bed_rows)})"
+    )
+    print(
+        f"Removed old (unique targets): {len(clinvar_old_removed)} ({len(old_to_remove_bed_rows)})"
+    )
+
+    return (new_to_add_bed_rows, old_to_remove_bed_rows)
+
+
+def get_bed_intersect(left_bed: str, right_bed: str) -> list[list[str]]:
+    not_in_bed_cmd = [
+        "bedtools",
+        "intersect",
+        "-a",
+        left_bed,
+        "-b",
+        right_bed,
+        "-v",
+    ]
+    result = subprocess.run(not_in_bed_cmd, capture_output=True, text=True, check=True)
+    intersected_regions = result.stdout.strip().splitlines()
+    # FIXME: This differs from the Perl script, and gives subtly differences in unique targets
+    # In this case 1270 vs 1274 (i.e. four are duplicated)
+    return [row.split("\t") for row in intersected_regions]
+    # return set(intersected_regions)
+
+
+def log_changes(
+    log_path: Path,
+    clinvar_old: dict[str, ClinVarVariant],
+    clinvar_new: dict[str, ClinVarVariant],
+    new_clinvar_to_add_keys: set[str],
+    old_clinvar_to_remove_keys: set[str],
+    new_benign: dict[str, ClinVarVariant],
+):
+    """
+    Log variants added and removed between clinvar versions
+    Check if part of latest ClinVar with benign status if removed
+    """
+    with log_path.open("w") as log_fh:
+
+        for new_key in new_clinvar_to_add_keys:
+
+            new_variant = clinvar_new.get(new_key)
+            if new_variant is None:
+                raise ValueError(f"New variant should be present for key {new_key}")
+
+            print(
+                f"ADDED: {new_variant.chrom}:{new_variant.pos}:{new_variant.CLNDN}:{new_variant.CLNSIG}",
+                file=log_fh,
+            )
+
+        for old_key in old_clinvar_to_remove_keys:
+            reason_for_removal = "MISSING"
+            if old_key in new_benign:
+                reason_for_removal = new_benign[old_key].CLNSIG
+            old_variant = clinvar_old.get(old_key)
+            if old_variant is None:
+                raise ValueError(f"New variant should be present for key {old_key}")
+            print(
+                f"REMOVED: {old_variant.chrom}:{old_variant.pos}:{old_variant.CLNDN}:{old_variant.CLNSIG} => {reason_for_removal}",
+                file=log_fh,
+            )
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
```

Comment:
You can pass a `description = "blah blah"` arg to `ArgumentParser` above with a nice little summary of what the script does. Please do.

Reply:
Yes, I'll add